File: src/Submission_In_Java/Lab1/Question1/Question1.java
package Submission_In_Java.Lab1.Question1;

import java.io.*;

public class Question1 {
    private static final PrintWriter pw;
    private static final BufferedReader br;

    static {
        try {
            pw = new PrintWriter(new BufferedOutputStream(new FileOutputStream("/home/denuvo-drm/Academic/Extra/Practice/src/Submission_In_Java/Lab1/Question1/output.txt")));
            br = new BufferedReader(new InputStreamReader(new FileInputStream("/home/denuvo-drm/Academic/Extra/Practice/src/Submission_In_Java/Lab1/Question1/input.txt")));
        } catch (FileNotFoundException e) {
            throw new RuntimeException(e);
        }
    }

    private static final int SIZE = (int) 1e2 + 340;
    private static final boolean[][] visited = new boolean[SIZE][SIZE];
    private static final char[][] graph = new char[SIZE][];
    private static int m = 0, n = 0;
    private static int curSize = 0;
    private static int maxSize = Integer.MIN_VALUE;

    public static void main(String[] args) throws IOException {
        // Take the input
        String inputLine = br.readLine();
        n = inputLine != null ? inputLine.replace(" ", "").length() : 0;
        while (inputLine != null) {
            graph[m] = inputLine.replace(" ", "").toCharArray();
            m++;
            inputLine = br.readLine();
        }
        // Done taking input ...
        br.close();
        // Time to dfs

        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                pw.print(graph[i][j]);
            }
            pw.println();
        }

        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (!visited[i][j] && graph[i][j] == 'Y') {
                    dfs(i, j);
                    maxSize = Integer.max(maxSize, curSize);
                    curSize = 0;
                }
            }
        }
        pw.println(maxSize);
        pw.close();
    }

    private static void dfs(int i, int j) {
        visited[i][j] = true;
        curSize++;
        // Time to check and call
        if (i > 0 && j > 0 && !visited[i - 1][j - 1] && graph[i - 1][j - 1] == 'Y') dfs(i - 1, j - 1);
        if (i > 0 && !visited[i - 1][j] && graph[i - 1][j] == 'Y') dfs(i - 1, j);
        if (i > 0 && j < n - 1 && !visited[i - 1][j + 1] && graph[i - 1][j + 1] == 'Y') dfs(i - 1, j + 1);
        if (j > 0 && !visited[i][j - 1] && graph[i][j - 1] == 'Y') dfs(i, j - 1);
        if (j < n - 1 && !visited[i][j + 1] && graph[i][j + 1] == 'Y') dfs(i, j + 1);
        if (i < m - 1 && j > 0 && !visited[i + 1][j - 1] && graph[i + 1][j - 1] == 'Y') dfs(i + 1, j - 1);
        if (i < m - 1 && !visited[i + 1][j] && graph[i + 1][j] == 'Y') dfs(i + 1, j);
        if (i < m - 1 && j < n - 1 && !visited[i + 1][j + 1] && graph[i + 1][j + 1] == 'Y') dfs(i + 1, j + 1);
    }
}

File: src/Submission_In_Python/Practice/Question1/Question1.py
import sys

sys.stdin = open("input.txt", 'r')

graph: [[str]] = []
while True:
    try:
        graph.append(input().split(" "))
    except EOFError:
        break

isVisited: [[bool]] = []
for _i in range(len(graph)):
    isVisited.append([])
    for second_index in range(len(graph[0])):
        isVisited[_i].append(False)

max_size: int = 0
cur_size: int = 0


def dfs(i: int, j: int):
    isVisited[i][j] = True
    global cur_size
    cur_size += 1
    # Time to check which node I can go
    if i > 0 and j > 0 and not isVisited[i - 1][j - 1] and \
            graph[i - 1][j - 1] == 'Y':
        dfs(i - 1, j - 1)
    if i > 0 and not isVisited[i - 1][j] and graph[i - 1][j] == 'Y':
        dfs(i - 1, j)
    if i > 0 and j < len(graph[0]) - 1 and not isVisited[i - 1][j + 1] and \
            graph[i - 1][j + 1] == 'Y':
        dfs(i - 1, j + 1)
    if j > 0 and not isVisited[i][j - 1] and graph[i][j - 1] == 'Y':
        dfs(i, j - 1)
    if j < len(graph[0]) - 1 and not isVisited[i][j + 1] and graph[i][j + 1] == 'Y':
        dfs(i, j + 1)
    if i < len(graph) - 1 and j > 0 and not isVisited[i + 1][j - 1] and \
            graph[i + 1][j - 1] == 'Y':
        dfs(i + 1, j - 1)
    if i < len(graph) - 1 and j < len(graph[0]) - 1 and not isVisited[i + 1][j + 1] and graph[i + 1][j + 1] == 'Y':
        dfs(i + 1, j + 1)
    if i < len(graph) - 1 and not isVisited[i + 1][j] and graph[i + 1][j] == 'Y':
        dfs(i + 1, j)


for _i in range(len(graph)):
    for second_index in range(len(graph[0])):
        if not isVisited[_i][second_index] and graph[_i][second_index] == 'Y':
            dfs(_i, second_index)
            max_size = max(cur_size, max_size)
            cur_size = 0

print(max_size)
